File: peteTheBaker.py
def cakes(recipe, available):
        output = []
        for k in recipe:
            if k in available:
                output.append(available[k] // recipe[k])
            else: return 0
        return min(output)

def cakes(recipe, available):
    return min(available.get(k, 0)//recipe[k] for k in recipe)

cakes({'cream': 200, 'flour': 300, 'sugar': 150, 'milk': 100, 'oil': 100}, {'sugar': 1700, 'flour': 20000, 'milk': 20000, 'oil': 30000, 'cream': 5000})

File: peteTheBaker.js
function cakes(recipe, available) {
    
    let availKeys = Object.keys(available)
    let recipeKeys = Object.keys(recipe)

    if (availKeys.includes(...recipeKeys)) {
        
        let ingredientCap = []

        for (const key in recipe) {
            ingredientCap.push(Math.floor(available[key] / recipe[key]))
        }

        return Math.min(...ingredientCap)
    }

    return 0

}

cakes({flour: 500, sugar: 200, eggs: 1}, {flour: 1200, sugar: 1200, eggs: 5, milk: 200})